catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,933 @@
+import * as M from '@material-ui/core'
+import { fade } from '@material-ui/core/styles'
+
+import * as APIConnector from 'utils/APIConnector'
+import AsyncResult from 'utils/AsyncResult'
+import * as AWS from 'utils/AWS'
+import * as Data from 'utils/Data'
+import Delay from 'utils/Delay'
+import * as NamedRoutes from 'utils/NamedRoutes'
+import StyledLink from 'utils/StyledLink'
+import pipeThru from 'utils/pipeThru'
+import * as s3paths from 'utils/s3paths'
+import { readableBytes } from 'utils/string'
+import tagged from 'utils/tagged'
+import useMemoEq from 'utils/useMemoEq'
+import * as validators from 'utils/validators'
+
+import * as PD from './PackageDialog'
+import * as ERRORS from './errors'
+import * as requests from './requests'
+
+const getNormalizedPath = (f) => (f.path.startsWith('/') ? f.path.substring(1) : f.path)
+
+const TYPE_ORDER = ['added', 'modified', 'deleted', 'unchanged']
+
+const useFilesInputStyles = M.makeStyles((t) => ({
+  root: {
+    marginTop: t.spacing(3),
+  },
+  header: {
+    alignItems: 'center',
+    display: 'flex',
+    height: 24,
+  },
+  headerFiles: {
+    ...t.typography.body1,
+    display: 'flex',
+  },
+  headerFilesDisabled: {
+    color: t.palette.text.secondary,
+  },
+  headerFilesError: {
+    color: t.palette.error.main,
+  },
+  headerFilesWarn: {
+    color: t.palette.warning.dark,
+  },
+  dropzoneContainer: {
+    marginTop: t.spacing(2),
+    position: 'relative',
+  },
+  dropzone: {
+    background: t.palette.action.hover,
+    border: `1px solid ${t.palette.action.disabled}`,
+    borderRadius: t.shape.borderRadius,
+    cursor: 'pointer',
+    display: 'flex',
+    flexDirection: 'column',
+    minHeight: 140,
+    outline: 'none',
+    overflow: 'hidden',
+  },
+  dropzoneErr: {
+    borderColor: t.palette.error.main,
+  },
+  dropzoneWarn: {
+    borderColor: t.palette.warning.dark,
+  },
+  active: {
+    background: t.palette.action.selected,
+  },
+  dropMsg: {
+    ...t.typography.body2,
+    alignItems: 'center',
+    display: 'flex',
+    flexGrow: 1,
+    justifyContent: 'center',
+    paddingBottom: t.spacing(1),
+    paddingTop: t.spacing(1),
+    textAlign: 'center',
+  },
+  dropMsgErr: {
+    color: t.palette.error.main,
+  },
+  dropMsgWarn: {
+    color: t.palette.warning.dark,
+  },
+  filesContainer: {
+    borderBottom: `1px solid ${t.palette.action.disabled}`,
+    maxHeight: 200,
+    overflowX: 'hidden',
+    overflowY: 'auto',
+  },
+  filesContainerErr: {
+    borderColor: t.palette.error.main,
+  },
+  filesContainerWarn: {
+    borderColor: t.palette.warning.dark,
+  },
+  added: {},
+  modified: {},
+  deleted: {},
+  unchanged: {},
+  fileEntry: {
+    alignItems: 'center',
+    background: t.palette.background.paper,
+    display: 'flex',
+    '&:not(:last-child)': {
+      borderBottomStyle: 'solid',
+      borderBottomWidth: '1px',
+      borderColor: 'inherit',
+    },
+    '&$added': {
+      background: M.colors.green[100],
```

Comment:
Maybe `t.palette.{success,warning,error}` are more appropriate

Reply:
they are too deep, and here we need something faint for the background